scripts/irods/test/test_resource_types.py

```diff
@@ -2385,6 +2385,153 @@ def test_auto_repl_on__2941(self):
         self.admin.assert_icommand("ils -L " + filename, 'STDOUT_SINGLELINE', 'cacheResc')
         self.admin.assert_icommand("ils -L " + filename, 'STDOUT_SINGLELINE', 'archiveResc')
 
+    def test_get_stale_cache_replica(self):
+
+        try:
+            filename = "testgetstalecachereplica.txt"
+            lib.make_file(filename, 10, 'arbitrary')
+            stale_replica_filename = "stalereplica.txt"
+            lib.make_file(stale_replica_filename, 20, 'arbitrary')
+
+            logical_path = os.path.join(self.admin.session_collection, filename)
+
+            self.admin.assert_icommand("iput " + filename)
+            self.admin.assert_icommand("iadmin modresc demoResc context \"auto_repl=off\"" )
+            self.admin.assert_icommand("iput -f " + stale_replica_filename + " " + filename)
```

Comment:
You don't specify a target resource on the calls to `iput`. Is that because the default resource is the compound resource?

Reply:
Yes, in this class the demoResc is changed to a compound resource.

https://github.com/irods/irods/blob/ee3a689971435d81e12aef6500d27ff33a9c2a01/scripts/irods/test/test_resource_types.py#L2003-L2006